src/facade/dragonfly_listener.cc

```diff
@@ -275,7 +272,14 @@ ProactorBase* Listener::PickConnectionProactor(LinuxSocketBase* sock) {
   }
 
   if (res_id == kuint32max) {
-    res_id = next_id_.fetch_add(1, std::memory_order_relaxed) % total;
+    uint32_t total = GetFlag(FLAGS_conn_io_threads);
+    uint32_t start = GetFlag(FLAGS_conn_io_thread_start) % pp->size();
+
+    if (total == 0 || total + start > pp->size()) {
+      total = pp->size() - start;
+    }
+
+    res_id = start + (next_id_.fetch_add(1, std::memory_order_relaxed) % total);
```

Comment:
If start is not set and total is less than the number of real threads, I'd shift it to make it use the tail (last half), as the last thread is not occupied by shards (by default). Also curious, why someone would choose not the tail for io threads

Reply:
Or do you expect conn_io_thread_start to be the more widely-used setting? Its a little confusing how they wrap, especially how you cut the total number off without warnings (I'd check-fail if the sums don't add up, so users could notice they got the settings wrong)

Reply:
It's not really a user-facing setting. It's a work in progress for us to have the flexibility to overlay Dragonfly on host threads but I hope at the end to have a more user-friendly heuristic.